metadata-ingestion/tests/unit/test_transform_dataset.py

```diff
@@ -262,7 +249,7 @@ def test_simple_dataset_ownership_transformation(mock_time):
     )
     assert first_ownership_aspect is None
 
-    last_event = outputs[3].record
+    last_event = outputs[2].record
```

Comment:
It is really hard for me to understand that test, even though I understand how transformers work and what happened. We should consider redesigning it. What I don't understand is why the even't with the ownership aspect for the dataset which is missing ownership aspect is now 3rd in the list and not 4th as before? I don't see how the change in test affected it.

Reply:
added test